index.js

```diff
@@ -23,12 +29,66 @@ function validate ({
       throw new Error('sampleInterval must be greather than eventLoopOptions.resolution')
     }
   }
+
   return {
     sampleInterval,
     eventLoopOptions
   }
 }
 
+class GCStats {
+  static entryKindToKey (entry) {
+    switch (entry.kind) {
+      case NODE_PERFORMANCE_GC_MAJOR:
+        return 'major'
+      case NODE_PERFORMANCE_GC_MINOR:
+        return 'minor'
+      case NODE_PERFORMANCE_GC_INCREMENTAL:
+        return 'incremental'
+      case NODE_PERFORMANCE_GC_WEAKCB:
+        return 'weakCB'
+      default:
+        return null
+    }
+  }
+
+  constructor () {
+    this.reset()
+  }
+
+  reset () {
+    // These are set as a tuple of [count; rollingAverage]
+    this.major = [0, undefined]
+    this.minor = [0, undefined]
+    this.incremental = [0, undefined]
+    this.weakCB = [0, undefined]
+  }
+
+  update (gcEntry) {
+    const key = GCStats.entryKindToKey(gcEntry)
+    if (key) {
+      let [count, average] = this[key]
+      if (average === undefined) {
+        this[key] = [1, gcEntry.duration]
+      } else {
+        count++
+        average -= average / count
+        average += gcEntry.duration / count
+        this[key] = [count, average]
+      }
```

Comment:
You used [this](https://en.wikipedia.org/wiki/Moving_average#Cumulative_moving_average) approach here, I like it.